intl: unexpose Intl.v8BreakIterator

bnoordhuis opened the pull request:
It was never an official Ecma-402 API, it is about to be superseded
by `Intl.Segmenter` and it's prone to crash under some circumstances.

Searches don't turn up any usage in the wild and the recommendation
from the V8 team is to remove it so now seems like a good a time as
any to do just that.

Fixes: https://github.com/nodejs/node/issues/8865
Fixes: https://github.com/nodejs/node/issues/14909
Refs: https://chromium-review.googlesource.com/c/v8/v8/+/620755

TimothyGu commented:
Does this *actually* fix #14909, when the property `delete` is in lib/internal/process.js, so the main context only?

TimothyGu commented on lib/internal/process.js:
```diff
@@ -133,20 +133,8 @@ function setupConfig(_source) {
-                                                   'DEP0017');
-    Object.defineProperty(Intl, 'v8BreakIterator', des);
-  }
+  if (typeof Intl !== 'undefined' && Intl.hasOwnProperty('v8BreakIterator'))
```
Is the hasOwnProperty check necessary?

bnoordhuis replied:
No.  I kept it to keep the logic more or less the same but I'll remove it.

bnoordhuis commented:
> Does this actually fix #14909

Sorry, should have been a `Refs:`

bnoordhuis commented:
@TimothyGu That did give me an idea on how to approach it more holistically from C++.  PTAL!

TimothyGu commented on src/node_internals.h:
```diff
@@ -121,6 +121,11 @@ inline v8::Local<TypeName> PersistentToLocal(
     v8::Isolate* isolate,
     const v8::Persistent<TypeName>& persistent);
 
+v8::Local<v8::Context> NewContext(
```
Some documentation would be helpful so that reader of the code would not need to look it up in node.cc.

bnoordhuis commented:
Updated with suggestions.  I also fixed a logic error in the test that was carried over from the old test because I wanted to keep the diff small...

CI: https://ci.nodejs.org/job/node-test-pull-request/9999/

addaleax commented:
Landed in 668ad449226940db92a5092822b77f3620631748

TimothyGu commented:
Just a question: has anyone tested that this works on no-Intl builds?
